test/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/nestMatesRedefinition/nmr001.java

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package com.ibm.jvmti.tests.nestMatesRedefinition;
+
+import com.ibm.jvmti.tests.util.Util;
+
+public class nmr001 {
+	/*
+	 * Redefines a class with given class bytes
+	 * 
+	 * @param	name			class being redefined
+	 * @param	classBytesSize	size of the new class bytes
+	 * @param 	classBytes		new class bytes
+	 * 
+	 * @return	true is redefinition successful; false otherwise
+	 */
+	public static native boolean redefineClass(Class name, int classBytesSize, byte[] classBytes);
+
+	/* 
+	 * Valid class definitions have the same nestmates attributes as the original
+	 * definition - either no nestmates attribute, a nest host attribute with the 
+	 * same nest host, or a nest members attribute with the same nest members in 
+	 * the same order.
+	 */
+	public boolean testRedefinitionNoNestAttributes() {
+		try {
+			boolean successfulRedefinition = false;
+			CustomClassLoader classloader = new CustomClassLoader();
+			byte[] bytes = ClassGenerator.nestHostNoAttribute();
+			Class<?> clazz = classloader.getClass("NestHost", bytes);
+			Object instance = clazz.getDeclaredConstructor().newInstance();
+			successfulRedefinition = redefineClass(clazz, bytes.length, bytes);
```

Comment:
Is this redefining the class to the same bytes?

Reply:
Yes - I assume that redefinition is valid because it does not change method signatures, nest attributes, or any other components which are disallowed from being changed.
Would using the same class bytes be expected to cause an issue (eg. redefinition notices that there is no change & does not redefine) that I did not see?
Should I remove this test case, since it is likely caught by the other class redefinition test cases, none of which have nestmates attributes?

Reply:
A clever JVM could treat this as a no-op and skip the redefinition.  We don't today but if we're clever enough, we might in the future.

A more robust test would modify the bytecodes to force the redefinition.